feat: pre-download Trivy vulnerability databases at build time

## Summary

Trivy's vulnerability DB and Java DB are not pre-downloaded during the container build. This means the first `trivy` scan at runtime requires downloading approximately 2.4 GB of databases, which is slow and fails in air-gapped environments.

## Proposed Change

Add `trivy image --download-db-only` and `trivy image --download-java-db-only` commands to the Dockerfile so the databases are cached at build time. This ensures scans work instantly without network access at runtime.

## Files

- `Dockerfile`

## CI Failure Report

**PR:** #1100
**Run:** 24931910563
**Workflow:** Docker Publish

### Failed Checks
- **smoke-test** failed during "Pull published image" step

### Log Output
```
failed to register layer: max depth exceeded
##[error]Process completed with exit code 1.
```

### Root Cause
The two new `RUN` instructions (`trivy image --download-db-only` and `trivy image --download-java-db-only`) added additional Docker layers that pushed the image past the overlay2 128-layer limit. The container image already operates near the layer cap.

### Action Required
The Trivy DB pre-download commands need to be combined into an existing `RUN` instruction (or combined into a single `RUN` with `&&`) to avoid adding extra layers. Alternatively, the commands could be appended to an adjacent `RUN` block in the Dockerfile to stay within the overlay2 128-layer limit.

Note: This repository previously encountered this exact issue — see commit `9ec4297` ("Revert fix: drop 23 layers from tail COPYs (under overlay2 128 cap)") and `54c4d6a` ("fix: drop 23 layers from tail COPYs (under overlay2 128 cap)").

--- a/Dockerfile
+++ b/Dockerfile
@@ -1447,12 +1447,10 @@ RUN npm exec chrome-devtools-mcp@0.20.2 -- --version 2>/dev/null || true
 # hadolint ignore=DL3059
 RUN npx -y oh-my-openagent install --no-tui \
     --claude=max20 --openai=no --gemini=no --copilot=no \
-    && rm -f ~/.config/opencode/*.bak.*
-
-# Pre-download Trivy vulnerability databases so first scan is instant.
-# Runs as vscode so cache lands in ~/.cache/trivy with correct ownership.
-# hadolint ignore=DL3059
-RUN trivy image --download-db-only --no-progress \
+    && rm -f ~/.config/opencode/*.bak.* \
+    # Pre-download Trivy vulnerability databases so first scan is instant.
+    # Runs as vscode so cache lands in ~/.cache/trivy with correct ownership.
+    && trivy image --download-db-only --no-progress \
     && trivy image --download-java-db-only --no-progress
 
 # ============================================================

